src/app/clusters/commodity-price-server/commodity-price-server.cpp

```diff
@@ -388,46 +371,128 @@ CHIP_ERROR Instance::SetCurrentPrice(const DataModel::Nullable<Structs::Commodit
     // Check PeriodStart  < PeriodEnd (if not null)
     if (!newValue.IsNull())
     {
-        Structs::CommodityPriceStruct::Type tempValue;
         if (!newValue.Value().periodEnd.IsNull() && (newValue.Value().periodStart > newValue.Value().periodEnd.Value()))
         {
             return CHIP_IM_GLOBAL_STATUS(ConstraintError);
         }
+
+        if (!newValue.Value().price.HasValue() && !newValue.Value().priceLevel.HasValue())
+        {
+            // Must have Price or PriceLevel
+            return CHIP_IM_GLOBAL_STATUS(ConstraintError);
```

Comment:
If this were part of a command invoke this would be INVALID_COMMAND, not CONSTRAINT_ERROR, since it's a conformance violation, not a constraint violation.

I guess this is just internal APIs?  If so, maybe it does not matter too much, or maybe we should just be returning an invalid value CHIP_ERROR instead of an IM status?